File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/picker_dialog_month/MonthSelection.java
package com.github.rooneyandshadows.lightbulb.dialogs.picker_dialog_month;

import com.github.rooneyandshadows.lightbulb.dialogs.base.BaseDialogSelection;

import java.util.Arrays;

class MonthSelection extends BaseDialogSelection<int[]> {
    public MonthSelection(int[] current, int[] draft) {
        super(current, draft);
    }

    @Override
    public boolean compareValues(int[] v1, int[] v2) {
        return Arrays.equals(v1, v2);
    }

    @Override
    public boolean hasCurrentSelection() {
        return getCurrentSelection() != null;
    }

    @Override
    public boolean hasDraftSelection() {
        return getDraftSelection() != null;
    }
}

File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/picker_dialog_month/MonthSelection.kt
package com.github.rooneyandshadows.lightbulb.dialogs.picker_dialog_month

import com.github.rooneyandshadows.lightbulb.dialogs.base.BaseDialogSelection
import java.util.*

internal class MonthSelection(current: IntArray?, draft: IntArray?) : BaseDialogSelection<IntArray?>(current, draft) {
    @Override
    override fun compareValues(v1: IntArray?, v2: IntArray?): Boolean {
        return Arrays.equals(v1, v2)
    }

    @Override
    override fun hasCurrentSelection(): Boolean {
        return getCurrentSelection() != null
    }

    @Override
    override fun hasDraftSelection(): Boolean {
        return getDraftSelection() != null
    }
}